File: app/src/main/java/com/example/qui__z/MainActivity.java
package com.example.qui__z;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;
import com.google.android.material.button.MaterialButton;

public class MainActivity extends AppCompatActivity {

    private String selectedTopic = "";
    private String selectedLevel = "";
    private String playerName = "";
    private CardView flagsCard, capitalsCard, nationalitiesCard, languagesCard, planetsCard;
    private MaterialButton startQuizBtn;

    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        playerName = getIntent().getStringExtra("USER_NAME");
        if (playerName == null || playerName.isEmpty()) {
            Toast.makeText(this, "Пожалуйста, введите ваше имя", Toast.LENGTH_SHORT).show();
            startActivity(new Intent(this, NameInputActivity.class));
            finish();
            return;
        }

        initializeViews();
        setupClickListeners();
    }

    private void initializeViews() {
        flagsCard = findViewById(R.id.flagsCard);
        capitalsCard = findViewById(R.id.capitalsCard);
        nationalitiesCard = findViewById(R.id.nationalitiesCard);
        languagesCard = findViewById(R.id.languagesCard);
        planetsCard = findViewById(R.id.planetsCard);
        startQuizBtn = findViewById(R.id.start_quiz_button);
    }

    private void setupClickListeners() {
        flagsCard.setOnClickListener(v -> selectCategory("flags"));
        capitalsCard.setOnClickListener(v -> selectCategory("capital"));
        nationalitiesCard.setOnClickListener(v -> selectCategory("nationality"));
        languagesCard.setOnClickListener(v -> selectCategory("language"));
        planetsCard.setOnClickListener(v -> selectCategory("planets"));

        startQuizBtn.setOnClickListener(v -> {
            if (!selectedTopic.isEmpty()) {
                Intent intent = new Intent(MainActivity.this, QuizActivity.class);
                intent.putExtra("CATEGORY", selectedTopic);
                intent.putExtra("USER_NAME", playerName);
                startActivity(intent);
            } else {
                Toast.makeText(MainActivity.this, "Пожалуйста, выберите категорию", Toast.LENGTH_SHORT).show();
            }
        });
    }

    private void selectCategory(String category) {
        selectedTopic = category;
        Toast.makeText(this, "Выбрана категория: " + category, Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/example/quiz/MainActivity.kt
package com.example.quiz

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.cardview.widget.CardView

class MainActivity : AppCompatActivity() {
    private var selectedQuizContainer: CardView? = null
    private var selectedQuizType: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        try {
            // Находим все контейнеры
            val quiz1Container = findViewById<CardView>(R.id.quiz1Container)
            val quiz2Container = findViewById<CardView>(R.id.quiz2Container)
            val quiz3Container = findViewById<CardView>(R.id.quiz3Container)
            val quiz4Container = findViewById<CardView>(R.id.quiz4Container)
            val quiz5Container = findViewById<CardView>(R.id.quiz5Container)
            val startButton = findViewById<Button>(R.id.startButton)

            // Создаем список контейнеров с их типами викторин
            val quizContainers = listOf(
                Pair(quiz1Container, "Викторина по флагам"),
                Pair(quiz2Container, "Викторина по столицам"),
                Pair(quiz3Container, "Викторина по национальностям"),
                Pair(quiz4Container, "Викторина по языкам"),
                Pair(quiz5Container, "Викторина по планетам")
            )

            // Устанавливаем обработчики нажатий для каждого контейнера
            quizContainers.forEach { (container, quizType) ->
                container.setOnClickListener {
                    // Сбрасываем цвет предыдущего выбранного контейнера
                    selectedQuizContainer?.setCardBackgroundColor(Color.WHITE)
                    (selectedQuizContainer?.getChildAt(0) as? android.widget.TextView)?.setTextColor(Color.BLACK)

                    // Устанавливаем черный цвет для выбранного контейнера
                    container.setCardBackgroundColor(Color.BLACK)
                    (container.getChildAt(0) as? android.widget.TextView)?.setTextColor(Color.WHITE)

                    // Сохраняем выбранный контейнер и тип викторины
                    selectedQuizContainer = container
                    selectedQuizType = quizType
                }
            }

            // Обработчик нажатия на кнопку "Начать викторину"
            startButton.setOnClickListener {
                if (selectedQuizType == null) {
                    Toast.makeText(this, "Пожалуйста, выберите викторину", Toast.LENGTH_SHORT).show()
                    return@setOnClickListener
                }

                val intent = Intent(this, QuizActivity::class.java).apply {
                    putExtra("QUIZ_TYPE", selectedQuizType)
                }
                startActivity(intent)
            }
        } catch (e: Exception) {
            Toast.makeText(this, "Ошибка при инициализации", Toast.LENGTH_SHORT).show()
        }
    }
} 
} 